lib/web_ui/lib/src/engine/canvaskit/image_filter.dart

```diff
@@ -109,3 +110,32 @@ class _CkBlurImageFilter extends CkImageFilter {
   }
 }
 
+class _CkMatrixImageFilter extends CkImageFilter {
+  _CkMatrixImageFilter({ required this.matrix, required this.filterQuality }) : super._();
+
+  Float64List matrix;
+  ui.FilterQuality filterQuality;
```

Comment:
Can these be `final`?

Reply:
Also, probably very minor performance wins, but if this object is opaque the values can be converted using `toSkMatrixFromFloat64` and `toSkFilterQuality` in the constructor initializer, so when the image filter is reused we don't need to re-convert.

Reply:
At first I was worried that this would change the toString() representation, but then I realized I could convert back for stringifying, so I went with this method.

Also, I wasn't copying the matrix so that was a problem waiting to happen. By converting the matrix to an Sk form, we have a defensive copy implicitly (although the stringify precision will suffer slightly, but that is arguably being honest about the precision that web maintains for these values...).

Reply:
In the end I made them final, but didn't store them as converted values in case that disrupts the processing of the == operator which is key to consistency of tree changes as values change.